sys/linux/test/bpf_helpers

```diff
@@ -20,3 +20,7 @@ r10 = bpf$MAP_CREATE_TAIL_CALL(AUTO, &AUTO={AUTO, AUTO, AUTO, AUTO, AUTO, 0x0, 0
 r11 = bpf$MAP_UPDATE_ELEM_TAIL_CALL(AUTO, &AUTO={{r10, r10}, &AUTO=0x0, &AUTO=r4, AUTO}, 0x1c)
 
 r4 = bpf$PROG_LOAD(AUTO, &AUTO={0x3, AUTO, &AUTO=@framed={{AUTO, AUTO, AUTO, AUTO, 0x0, AUTO, AUTO, AUTO, 0x0}, [@tail_call={{AUTO, AUTO, AUTO, AUTO, r10, AUTO, AUTO, AUTO, AUTO}, {AUTO, AUTO, AUTO, AUTO, AUTO, AUTO, AUTO}, {AUTO, AUTO, AUTO, AUTO}, {AUTO, AUTO, AUTO, AUTO, AUTO, AUTO, AUTO}}], {AUTO, AUTO, AUTO, AUTO}}, &AUTO='GPL\x00', 0x0, 0x0, 0x0, 0x0, 0x0, "00000000000000000000000000000000", 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, 0xa0)
+
+# Prepare, load, and execute a BPF program that calls ringbuf helpers
```

Comment:
Please keep in mind that syzkaller has limits on the maximum number of calls in a single program (and the test is executed as a single program).

So if the file covers different functionality, it's better to split it.

Reply:
Aha! I didn't know that. Thanks for the heads up!

Do you remember what the limit is? There are more than 200 BPF helpers and I'd like to cover most of them with such tests (though it will take time), so I probably want to avoid doing one file per helper :grimacing: I'll probably split differently if the limit is 10 calls vs. 50.

Reply:
`prog.MaxCalls` is 40

Reply:
Ok. I can probably split it by required program types (so far I only added helpers that can be used from all program types) and then maybe move the tests for map helpers to their own file since there are quite a few. My intuition is that helpers for the same program type are maybe more likely to share the same required preparation syscalls (to prepare maps or kernel state for example).

It will probably be clearer as we add more helpers.

(Unrelated: I'm also considering splitting `bpf.txt` to have all instructions and helper descriptions in their own file. That file is becoming huge.)